packages/flutter_tools/bin/xcode_debug.js

```diff
@@ -0,0 +1,519 @@
+  }
+
+  const xcodeResult = getXcode(args);
+  if (xcodeResult.error != null) {
+    return new RunJsonResponse(false, xcodeResult.error).stringify();
+  }
+  const xcode = xcodeResult.result;
+
+  if (args.command === 'project-opened') {
+    const result = getWorkspace(xcode, args);
+    return new RunJsonResponse(result.error == null, result.error).stringify();
+  } else if (args.command === 'debug') {
+    const result = debugApp(xcode, args);
+    return new RunJsonResponse(result.error == null, result.error, result.result).stringify();
+  } else if (args.command === 'stop') {
+    const result = stopApp(xcode, args);
+    return new RunJsonResponse(result.error == null, result.error).stringify();
+  } else {
+    return new RunJsonResponse(false, 'Unknown command').stringify();
+  }
+}
+
+/**
+ * Parsed and validated arguments passed from the command line.
+ */
+class CommandArguments {
+  /**
+   *
+   * @param {!Array<string>} args List of arguments passed from the command line.
+   */
+  constructor(args) {
+    this.command = this.validatedCommand(args[0]);
+
+    const parsedArguments = this.parseArguments(args);
+
+    this.xcodePath = this.validatedStringArgument('--xcode-path', parsedArguments['--xcode-path']);
+    this.projectPath = this.validatedStringArgument('--project-path', parsedArguments['--project-path']);
+    this.workspacePath = this.validatedStringArgument('--workspace-path', parsedArguments['--workspace-path']);
+    this.targetDestinationId = this.validatedStringArgument('--device-id', parsedArguments['--device-id']);
+    this.targetSchemeName = this.validatedStringArgument('--scheme', parsedArguments['--scheme']);
+    this.skipBuilding = this.validatedBoolArgument('--skip-building', parsedArguments['--skip-building']);
+    this.launchArguments = this.validatedJsonArgument('--launch-args', parsedArguments['--launch-args']);
+    this.closeWindowOnStop = this.validatedBoolArgument('--close-window', parsedArguments['--close-window']);
+    this.promptToSaveBeforeClose = this.validatedBoolArgument('--prompt-to-save', parsedArguments['--prompt-to-save']);
+    this.verbose = this.validatedBoolArgument('--verbose', parsedArguments['--verbose']);
+
+    if (this.verbose === true) {
+      console.log(JSON.stringify(this));
+    }
+  }
+
+  /**
+   * Validates the command is available.
+   *
+   * @param {?string} command
+   * @returns {!string} The validated command.
+   * @throws Will throw an error if command is not recognized.
+   */
+  validatedCommand(command) {
+    switch (command) {
+      case 'project-opened':
+      case 'debug':
+      case 'stop':
+        return command;
+      default:
+        throw `Unrecognized Command: ${command}`;
+    }
+  }
+
+  /**
+   * Validates the flag is allowed for the current command.
+   *
+   * @param {!string} flag
+   * @returns {!bool} The validated command.
+   */
+  isArgumentAllowed(flag) {
```

Comment:
Should we be crashing if a not allowed argument is passed? Wouldn't this suggest that someone changed the tool dart library that invokes this but forgot to update this js script?

Reply:
oh wait, looks like you do that on line 84?

Reply:
I updated it to crash when using a not allowed argument